obelisk/client.py

```diff
@@ -74,6 +77,16 @@ def __init__(self, address, public_key=None, log=None, heartbeat_port=9092):
         self.listen_heartbeat(heartbeat_port)
         task.LoopingCall(self.renew_subscriptions).start(120, now=False)
 
+    def _reconnect(self):
+        if self.addresses is not None:
+            try:
+                index = self.addresses.index(self.address)
+                index = (index + 1) % len(self.addresses)
+                self.address = self.addresses[index]
+            except ValueError:
+                pass
+        return ClientBase._reconnect(self)
+
     def listen_heartbeat(self, port):
         def timeout():
             self.connected = False
```

Comment:
The commit here does that:
https://github.com/OpenBazaar/python-libbitcoinclient/pull/3/commits/be16b5301796a99e6abac2bf97a086772c71e206

Is there every a scenario where _reconnect occurs and a heartbeat timeout? It wouldn't be ideal if it rotated servers twice.